### PR TITLE
Feature/add cmake flag skip grpc lib compile

### PR DESCRIPTION
**Pull request by eitanhs**
Adressing @chwarr request - see https://github.com/microsoft/vcpkg/pull/10319#discussion_r425347394

I hope I got it right, I'm not very familiar with CMake syntax...

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -11,6 +11,9 @@ set (CMAKE_MODULE_PATH
 set (BOND_ENABLE_GRPC
     "TRUE"
     CACHE BOOL "If FALSE, then do not build gRPC integration")
+set (BOND_SKIP_GRPC_LIB_COMPILE
+    "FALSE"
```
Comment (bhardwajs) [inline CMakeLists.txt:15]: @chwarr Should we set this to be default to `TRUE` and `BOND_ENABLE_GRPC` to `FALSE`?

File: thirdparty/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 include (Compiler)
 
-if (BOND_ENABLE_GRPC)
+if (BOND_ENABLE_GRPC AND NOT BOND_SKIP_GRPC_LIB_COMPILE)
```
Comment (bhardwajs) [inline thirdparty/CMakeLists.txt:3]: Doesn't this mean both `BOND_ENABLE_GRPC` should be enabled and `BOND_SKIP_GRPC_LIB_COMPILE` be disabled?

Seems like the best thing to do would be to set `BOND_ENABLE_GRPC` to `FALSE` and let the user opt-in?

File: CMakeLists.txt (hunk 1 above)
Comment (chwarr) [reply CMakeLists.txt:15]: That would be a breaking change. I'm not sure the benefit is large enough. I have a different approach that sort of avoids this.

**Comment by chwarr** [comment]
In https://github.com/microsoft/bond/commit/67eec236eb87f59b76a98ed21ffe812cd09d3cdf, support was added to find RapidJSON via `find_package`. I think that something similar would work here, and it avoids some of the issues that @bhardwajs mentioned. I think we want something similar here.

**Comment by chwarr** [comment]
@eitanhs, are you still interested in pursuing this PR? If not, are you OK closing it?

**Comment by bhardwajs** [comment]
> @eitanhs, are you still interested in pursuing this PR? If not, are you OK closing it?

And, @eitanhs, if you are not interested in pursuing, please let me know and I can take it over. Thanks!

**Comment by eitanhs** [comment]
> > @eitanhs, are you still interested in pursuing this PR? If not, are you OK closing it?
> 
> And, @eitanhs, if you are not interested in pursuing, please let me know and I can take it over. Thanks!

Go for it! Good luck!

**Comment by bhardwajs** [comment]
@chwarr @eitanhs - I wasn't sure how to push changes to this PR. So, I ended up creating #1097. Thanks!

**Comment by chwarr** [comment]
Cool. I'll close this in favor of #1097 then.

**Pull request closed without merging**